internal/tokens/https/https.go

```diff
@@ -136,24 +136,28 @@ func (h *HTTPS) handleTokenPoolCreate(ctx context.Context, data fftypes.JSONObje
 		return nil // move on
 	}
 
-	hexUUIDs, err := hex.DecodeString(clientID)
-	if err != nil || len(hexUUIDs) != 32 {
-		log.L(ctx).Errorf("TokenPool event is not valid - bad uuids (%s): %+v", err, data)
+	unpackedData := createPoolData{}
+	err = json.Unmarshal([]byte(packedData), &unpackedData)
```

Comment:
Because the data field is arbitrary, is there any way someone could craft something that would be unpacked in an unexpected way here? For example, could someone override ID, TransactionID, etc?

Reply:
Short answer: no, I don't think it's an issue.

Longer answer: a lot of it has to do with the network architecture and which channels are considered private/immune to man-in-the-middle attacks. I'd assume we prevent direct access to the token connector, meaning the only way to "craft" data would be to actually write it on chain. In that case, if it matches the expected schema, it is a valid transaction for all intents and purposes, and all FireFly nodes listening to the stream would pick up and process it.

We do most of our validation at the db plugin layer just before upsert, and we do protect there against invalid things like changing immutable fields (namespace/name) once they've been recorded once.